src/Toml/Parser.hs

```diff
@@ -162,7 +162,11 @@ tableNameP = lexeme $ between (char '[') (char ']') keyP
 integerP :: Parser Integer
 integerP = lexeme $ binary <|> octal <|> hexadecimal <|> decimal
   where
-    decimal      = L.signed sc L.decimal
+    decimal      = L.signed sc (noLeadingZero <|> zero)
```

Comment:
Looks like you didn't update on latest master, there's different `decimalP` parser is used:

* https://github.com/kowainik/tomland/blob/3187fed8dc353b68553f229b5dfb76fdc4e5e9f0/src/Toml/Parser.hs#L176

I think you should update `decimalP` parser directly to forbid leading zeroes. Otherwise the implementation looks good 👍